homeassistant/components/jewish_calendar/config_flow.py

```diff
@@ -28,32 +29,40 @@ class JewishCalendarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return JewishCalendarOptionsFlowHandler(config_entry)
+
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
+        errors = {}
+
         if user_input is not None:
-            return self.async_create_entry(
-                title=user_input.get(CONF_NAME, DEFAULT_NAME),
-                data={
-                    CONF_NAME: user_input.get(CONF_NAME, DEFAULT_NAME),
-                    CONF_DIASPORA: user_input.get(CONF_DIASPORA, DEFAULT_DIASPORA),
-                    CONF_LANGUAGE: user_input.get(CONF_LANGUAGE, DEFAULT_LANGUAGE),
-                    CONF_CANDLE_LIGHT_MINUTES: user_input.get(
-                        CONF_CANDLE_LIGHT_MINUTES, DEFAULT_CANDLE_LIGHT
-                    ),
-                    CONF_HAVDALAH_OFFSET_MINUTES: user_input.get(
-                        CONF_HAVDALAH_OFFSET_MINUTES, DEFAULT_HAVDALAH_OFFSET_MINUTES
-                    ),
-                    CONF_LATITUDE: user_input.get(
-                        CONF_LATITUDE, self.hass.config.latitude
-                    ),
-                    CONF_LONGITUDE: user_input.get(
-                        CONF_LONGITUDE, self.hass.config.longitude
-                    ),
-                },
-            )
+
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
 
         return self.async_show_form(
-            step_id="user", data_schema=vol.Schema(DATA_SCHEMA), errors={}
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+                    vol.Optional(CONF_DIASPORA, default=DEFAULT_DIASPORA): bool,
+                    vol.Optional(CONF_LANGUAGE, default=DEFAULT_LANGUAGE): vol.In(
+                        ["hebrew", "english"]
+                    ),
+                }
+            ),
+            errors={},
```

Comment:
We should show the stored errors here.

Reply:
Right, missed that. I used transmission client as a template and forgot I brought back the errors.